### PR TITLE
IPEndPoint.Port as UInt16

### PR DESCRIPTION
**Pull request by deinok**
A Port is a unsigned integer of 16 bits, so it have sense to define it internaly as a UInt16

### CODE REVIEW

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs [hunk 1]
```diff
@@ -20,23 +20,23 @@ public class IPEndPoint : EndPoint
...
         private IPAddress _address;
-        private int _port;
+        private UInt16 _port;
 
         internal const int AnyPort = MinPort;
 
-        internal static IPEndPoint Any = new IPEndPoint(IPAddress.Any, AnyPort);
-        internal static IPEndPoint IPv6Any = new IPEndPoint(IPAddress.IPv6Any, AnyPort);
+        internal static readonly IPEndPoint Any = new IPEndPoint(IPAddress.Any, AnyPort);
+        internal static readonly IPEndPoint IPv6Any = new IPEndPoint(IPAddress.IPv6Any, AnyPort);
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: Are these actually used anywhere?  If not, they can instead be deleted.

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs [hunk 2]
```diff
@@ -20,23 +20,23 @@ public class IPEndPoint : EndPoint
...
         ///       Specifies the maximum acceptable value for the <see cref='System.Net.IPEndPoint.Port'/>
         ///       property.
         ///    </para>
         /// </devdoc>
-        public const int MaxPort = 0x0000FFFF;
+        public const int MaxPort = UInt16.MaxValue;
 
         private IPAddress _address;
-        private int _port;
+        private UInt16 _port;
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: short

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public IPEndPoint(long address, int port)
             {
                 throw new ArgumentOutOfRangeException(nameof(port));
             }
-            _port = port;
+            _port = Convert.ToUInt16(port);
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: This should just be `(ushort)port`.  It was already validated to be in range by ValidatePortNumber, and this Convert call just bloats things unnecessarily.  Same for the rest of the Convert calls below: they should just be casts.

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs [hunk 4]
```diff
@@ -20,23 +20,23 @@ public class IPEndPoint : EndPoint
...
+        public const int MinPort = UInt16.MinValue;
 
         /// <devdoc>
         ///    <para>
         ///       Specifies the maximum acceptable value for the <see cref='System.Net.IPEndPoint.Port'/>
         ///       property.
         ///    </para>
         /// </devdoc>
-        public const int MaxPort = 0x0000FFFF;
+        public const int MaxPort = UInt16.MaxValue;
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: ushort

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs [hunk 5]
```diff
@@ -20,23 +20,23 @@ public class IPEndPoint : EndPoint
         ///       property.
         ///    </para>
         /// </devdoc>
-        public const int MinPort = 0x00000000;
+        public const int MinPort = UInt16.MinValue;
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: ushort

**Comment by wfurt** [comment]
What is motivation for this change @deinok ? It seems like we will now have more conversions and I'm not sure if you expect visible savings in memory usage.

**Comment by stephentoub** [comment]
> I'm not sure if you expect visible savings in memory usage

I don't expect there would be any.

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs (hunk 1 above)
Comment (deinok) [reply src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: IDK, a simple compilation test should indicate if they are required, so I will delete it, and if it fails i will revert it.

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs (hunk 2 above)
Comment (deinok) [reply src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: Done

File: src/System.Net.Primitives/src/System/Net/IPEndPoint.cs (hunk 3 above)
Comment (deinok) [reply src/System.Net.Primitives/src/System/Net/IPEndPoint.cs]: Done

**Comment by deinok** [comment]
@wfurt My motivation is that it have for me more sense to represent a port with an `ushort`. I don't expect performace improvements, only 2 byte saving that it's not too much.

Idealy the exposed Port property should be an ushort but better not change the API design.
Better have compilation checkings than use TcpValidationHelpers.ValidatePortNumber(value) to check if its exactly in the range of an `ushort`

**Comment by davidsh** [comment]
cc: @karelz

@deinok Thx for the PR.  At this point, with all of iterations to the PR, the change is basically just to storing the internal port value as a `ushort` and not `int`.  We can't change the public APIs of the parameters or public const values.  It seems that the memory savings of this change aren't really worth the added weight of readability and maintainability of the code.

**Comment by deinok** [comment]
Yeah, this would be good if we can change the API, so I close this ;)

**Pull request closed without merging**